src/Shared/RoslynUtils/SymbolExtensions.cs

```diff
@@ -145,4 +186,17 @@ public static bool TryGetNamedArgumentValue<T>(this AttributeData attribute, str
         }
         return false;
     }
+
+    public static string GetParameterInfoFromConstructor(this IParameterSymbol parameterSymbol)
+    {
+        if (parameterSymbol is { ContainingSymbol: IMethodSymbol constructor })
+        {
+            var parameterType = $"typeof({parameterSymbol.Type.ToDisplayString()})";
```

Comment:
Remind me what `ToDisplayString` does if the type is something like `class @class { }`?

Reply:
It looks like the `@` is stripped from the type symbol's identifier.

Reply:
That sounds bad.  Won't you send up with `typeof(class)`, which is a syntax error?

Reply:
Ah! I misinterpreted your comment a little bit. I thought your inquiry was around the use of the `@` prefix on class names in general (which ends up producing something like `typeof(MyClass)` which is valid) but you were referring to the literal use of `class @class`.

My initial reaction to this is that it's the kind of edge case that I'd be comfortable not addressing specifically in RDG.

My second reaction is to explore what the implementation complexity would look like with this and see if it is worth pursuing given cost/reward. It might very well be we have to pass a formatter to `ToDisplayString` here to get the desired behavior.

My third reaction is wondering if the current behavior (compiler errors on generated code) is a valid experience for users who run into these kinds of buggy experiences.

In any case, this is worth tracking in a follow-up issue if you'd like to file one?

Reply:
I would expect this to come up in a large variety of source generators, so I would guess there's a standard solution (maybe a DisplayString variant or an escaping API).